### ISSUE TITLE
[Enhancement] Enable declarative list building via projected content

### ISSUE THREAD

**Issue by RasmusKjeldgaard** [kirbydesign/designsystem#1813]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
To provide better control over list, I would like to be able to have a list version where I do not supply data to the list via the `items` input, but instead via content projection, where I declare the list something like this: 

```
<kirby-list>
    <kirby-item>
        ...
    </kirby-item>

    <kirby-item>
        ...
    </kirby-item>

    <kirby-item>
        ...
    </kirby-item>
</kirby-list>
```
This requirement originates from experiments with virtual scroll described in #1794, where it would suit the use-case better if consuming projects could have more fine grained control over their lists, so they can add necessary html and css to e.g. the items, to supply the correct information of item heights and similar to the virtual scroll algorithms.

Along these lines, this approach will give most likely make for a more flexible list, that can support many more use-cases, without having to account for these directly in the kirby source code, as teams themselves can make small adjustments.

This approach is also in line with the standard HTML approach, [ionic](https://ionicframework.com/docs/api/list), and how [Material Design](https://material.io/components/lists/web#single-line-list) and [Carbon](https://material.io/components/lists/web#single-line-list) does it.

## Describe the solution you'd like
This solution should be strictly limited to adding an option for content projection to the list, as #1814  and #1815 will add proper styling, and #1816 will handle sections. This could be as simple as using the existing template if the items input has been set, and using a new `<ng-content></ng-content>` template if items is not set. 

## Have you considered any alternatives?
We could consider to keep the current structure, and try to implement use cases like virtual scrolling ourselves, continuing the work on #1606. 

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

**Comment by MadsBuchmann** [comment]
We should consider how we want to do this. Could probably use some tech refinement. 

Do we want to split it up into two components? One for the list projecting content and one for the list using input properties. The list that uses projection could be created as a wholly new component (with HEAVY inspiration from the old list) in the core package. This would require us to also move the `Kirby-item` to the core package. But it is rather thin and i suspect it should not be a big issue. 

We should do a preliminary investigation if this would cause any issues.

**Comment by jkaltoft** [comment]
Yes, let's do some tech refinement on this.

A few questions comes to mind, among others:

- Is this supposed to be a compound component? 
    - `<kirby-list-item>` is not supposed to be used outside `<kirby-list>`
    - Only `<kirby-list-item>` are allowed inside `<kirby-list>`
- If so, is this the first truly compound component in Kirby or do we have other examples?
- If not, what is the difference between using a `<kirby-list-item>` and another component, like `<kirby-item>`?
- What about nesting `<kirby-list>` inside `<kirby-list-item>`?

**Comment by MadsBuchmann** [comment]
Is this replaced by #1883?
